### ISSUE TITLE
VPECERR override in BBC functionality using submit_batch_jobs.sh

### ISSUE THREAD

**Issue by erikpeterson23** [RickKessler/SNANA#733]
Working with Dan Scolnic on various peculiar velocity treatments.

I’m trying to run `submit_batch_jobs.sh` on a `.input` file with a ton of `datafile_override` files like so:
```  
...
MUOPT:
  - 'datafile_override=/project2/rkessler/SURVEYS/DES/USERS/epeterson3/pec_vel/header_overrides/bulk/lowz_zero.txt'
  - 'datafile_override=/project2/rkessler/SURVEYS/DES/USERS/epeterson3/pec_vel/header_overrides/bulk/pos_lowz_25.txt'
  - 'datafile_override=/project2/rkessler/SURVEYS/DES/USERS/epeterson3/pec_vel/header_overrides/bulk/pos_lowz_50.txt'
...
```
In each of these `datafile_override` files, I have originally had these columns: `VARNAMES: CID zHEL VPEC`, but I’d like to add a column for `VPECERR` as well. I tried `VPEC_ERR` as the column name and that failed, and then I tried `VPECERR` as the column name and that ran successfully, but that didn’t update the `VPECERR` column. Do you have an idea on what I can do to fix this?

I have copied the `.input` file to `$SNANA_DEBUG/epeterson3` along with a sample `datafile_override` file with the desired `VPECERR` column. I run my fitting procedure with `submit_batch_jobs.sh plus_erik_9_20_21.input`

**Comment by erikpeterson23** [comment]
As requested from Rick, I copied one of the faulty `CPU*.CMD` files to `$SNANA_DEBUG/epeterson3`

The resulting interactive job that results in errors/issues is this:
```
SALT2mu.exe plus_erik_9_20_21.input      prefix=datafile_arg_FITOPT000_MUOPT065      write_yaml=1    datafile_override=/project2/rkessler/SURVEYS/DES/USERS/epeterson3/pec_vel/header_overrides/bulk/darcy_sdss_vpec_var.txt   &>  datafile_arg_FITOPT000_MUOPT065.LOG
```
(I have copied the `datafile_override` file to `$SNANA_DEBUG/epeterson3` as well)

When I run this command with `VPECERR` as the column name in my `datafile_override` file, `SALT2mu.exe` fails. The .LOG file for that error is called `datafile_arg_FITOPT000_MUOPT065_VPECERR.LOG`, and it says:
```
 FATAL ERROR ABORT called by read_data_override
   Unable to implement override for VPECERR
   Might need to update function read_data_override
```

When I run this command with `VPEC_ERR` as a column name, `SALT2mu.exe` runs successfully, but the column `VPECERR` in the outputted `.FITRES` file does not have updated values -- all `VPECERR`s remain at 250.

**Comment by RickKessler** [comment]
fixed override logic to work when two override variables  (e..g, zHEL and VPEC) both modify zHD.

**Comment by erikpeterson23** [comment]
`VPECERR` is not being propagated to `MUERR`. 

I compared the `MUERR` values for two files with the same information but with only `VPECERR` modified (`datafile_override` files copied to `$SNANA_DEBUG/epeterson3`: `darcy_sdss_vpec.txt` (with constant `VPECERR`) and `darcy_sdss_vpec_var.txt` (with variable `VPECERR`)), and we found no change in `MUERR` (see the attached plot and/or the resulting `FITRES` files also copied to `$SNANA_DEBUG/epeterson3` -- constant `VPECERR`: `FITOPT000_MUOPT063.FITRES.gz`, variable `VPECERR`: `FITOPT000_MUOPT065.FITRES.gz`).

Commands to test either of these in `$SNANA_DEBUG/epeterson3` are as follows: constant `VPECERR`: `RICK_TEST`, variable `VPECERR`: `RICK_TEST_var`.

![variable_vpec_muerr_compare_z](https://user-images.githubusercontent.com/52798367/142648551-32146606-16e6-4452-9f6a-1d7a777aeb18.png)
